src/protocols/secure_channel/CASEServer.cpp

```diff
@@ -123,6 +124,9 @@ CHIP_ERROR CASEServer::OnMessageReceived(Messaging::ExchangeContext * ec, const
 
 void CASEServer::PrepareForSessionEstablishment(const ScopedNodeId & previouslyEstablishedPeer)
 {
+    chip::timing::DurationTimer timer = chip::timing::GetDefaultTimingInstance( "Inet: CASE Session establishment" );
+    timer.start();
```

Comment:
generally you want RAII rather than start/stop.

Although since this happens across messages you probably need something else. BEGIN/END exist in tracing too, however you must make sure they are properly nested by contract (not sure if that can be done here) or you could use instant tracing and then post-process for getting durations.

Reply:
Actually looking at code these are always local variables, so TRACE_SCOPE works fine here. and is RAII (which this code is not and should be)

Reply:
Thanks @andy31415 and @bzbarsky-apple for the review. Your comments are much appreciated. I will look at extending TRACE_SCOPE.